sktime/classification/compose/tests/test_feature_importances_.py

```diff
@@ -0,0 +1,85 @@
+
+
+# Compare simple cases of feature importances results obtained
+# directly from TimeSeriesForestClassifier with pipeline implementation
+# on the same intervals using the same DecisionTreeClassifier
+
+@pytest.mark.parametrize("n_intervals", [1, 2])
+@pytest.mark.parametrize("n_estimators", [1, 2])
+def test_feature_importances_(n_intervals, n_estimators):
+
+    random_state = 1234
+    n_features = 3
+    X_train, y_train = load_gunpoint(split="TRAIN", return_X_y=True)
+
+    # Compute feature importances using the default method
+    features = [np.mean, np.std, time_series_slope]
+    steps = [('transform', RandomIntervalFeatureExtractor(
+                n_intervals=n_intervals,
+                features=features,
+                random_state=random_state)),
+             ('clf', DecisionTreeClassifier())]
+    base_estimator = Pipeline(steps)
+    clf1 = TimeSeriesForestClassifier(estimator=base_estimator,
+                                      random_state=random_state,
+                                      n_estimators=n_estimators)
+    clf1.fit(X_train, y_train)
+
+    fi1 = np.zeros([n_estimators, n_intervals*n_features])
+    fi2 = np.zeros([n_estimators, n_intervals*n_features])
+
+    # Obtain intervals and decision trees from fitted classifier
+    for i in range(n_estimators):
+        intervals = clf1.estimators_[i].steps[0][1].intervals_
+        steps = [
+            ('segment', IntervalSegmenter(intervals)),
+            ('transform', FeatureUnion([
+                ('mean', RowTransformer(
+                    FunctionTransformer(func=np.mean, validate=False))),
+                ('std', RowTransformer(
+                    FunctionTransformer(func=np.std, validate=False))),
+                ('slope', RowTransformer(
+                    FunctionTransformer(func=time_series_slope,
+                                        validate=False)))
+                ])),
+            ('clf', clone(clf1.estimators_[i].steps[-1][1]))
+        ]
+        clf2 = Pipeline(steps)
+        clf2.fit(X_train, y_train)
+
+        # Compute and check for individual feature importances
+        fi1[i, :] = clf1.estimators_[i].steps[-1][1].feature_importances_
+        fi2[i, :] = clf2.steps[-1][1].feature_importances_
+        np.testing.assert_array_equal(fi1[i, :], fi2[i, :])
+
+    # Compute and check for final feature importances
+    fis1 = clf1.feature_importances_
+
+    n_timepoints = len(clf1.estimators_[0].steps[0][1]._time_index)
+    fis2 = np.zeros((n_timepoints, n_features))
+
+    for i in range(n_estimators):
+        intervals = clf1.estimators_[i].steps[0][1].intervals_
+        for j in range(n_features):
+            for k in range(n_intervals):
+                start, end = intervals[k]
+                fis2[start:end, j] += fi2[i, (j * n_intervals) + k]
+    fis2 = fis2 / n_estimators / n_intervals
+    np.testing.assert_array_equal(fis1, fis2)
```

Comment:
is there a case when this one could be true but the first assert statement (line 70) false? Otherwise we could only have the first one, no?

Reply:
My idea was that the first assert statement would show the individual feature importances are correct, and in the event where the first is true while the second is false, we would be certain that the error is caused in collating the final resutls for the entire time series.